File: src/test/java/com/github/bzumhagen/sct/ChangelogConfigurationTest.java
package com.github.bzumhagen.sct;

import org.junit.Test;

import java.util.HashSet;
import java.util.Set;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class ChangelogConfigurationTest {

    @Test
    public void defaultTags() {
        ChangelogConfiguration defaultConfig = new ChangelogConfiguration();
        Set<String> expectedTags = new HashSet<String>();
        expectedTags.add("added");
        expectedTags.add("removed");
        expectedTags.add("changed");
        expectedTags.add("deprecated");

        assertEquals(defaultConfig.getTags(), expectedTags);
    }
}

File: src/test/scala/com/github/bzumhagen/sct/ChangelogConfigurationTest.scala
package com.github.bzumhagen.sct

import java.util.regex.Pattern

import org.scalatest.{FlatSpec, Matchers}

class ChangelogConfigurationTest extends FlatSpec with Matchers {

  "ChangelogConfiguration" should "load the default configuration" in {
    val config = ChangelogConfiguration.load()
    val expectedName = "SCT"
    val expectedVersionPattern = "[\\s\\S]*\nversion: (.+)[\\s\\S]*".r
    val expectedTagPattern = "[\\s\\S]*\ntag: (.+)[\\s\\S]*".r
    val expectedReferencePattern = "[\\s\\S]*\nresolves: (.+)[\\s\\S]*".r
    val expectedTags = Set(
      "Added",
      "Changed",
      "Removed",
      "Deprecated"
    )

    config.name shouldBe expectedName
    config.versionPattern.regex shouldBe expectedVersionPattern.regex
    config.tagPattern.regex shouldBe expectedTagPattern.regex
    config.referencePattern.regex shouldBe expectedReferencePattern.regex
    config.tags shouldBe expectedTags
    config.smartGrouping shouldBe true
  }
}
